language/service/src/xlr/registry.ts

```diff
@@ -36,15 +36,15 @@ export class PlayerxlrRegistry extends BasicXLRRegistry {
 
   add(type: NamedType<NodeType>, plugin: string, capability: string): void {
     let registeredName = type.name;
-
-    // Figure out how the Asset/View will be referenced
+    // Figure out how the Asset/View will be referenced from the type argument that will fill in Asset
     if (
       (capability === 'Assets' || capability === 'Views') &&
       type.type === 'object' &&
-      type.properties?.type?.node?.type === 'string' &&
-      type.properties.type.node.const
+      type.extends?.genericArguments?.[0].type === 'string' &&
```

Comment:
is `genericArguments` guaranteed to be a non-empty array if its not `undefined` ?

Reply:
From the XLRs we generate it will be either undefined or an array with at least one item in it but I will add a optional chain accessor to that call. Also since we are accessing something that claims to be an `Asset` or a `View` it should extend `Asset` and if it doesn't it doesn't conform to the "latest" XLR spec.